src/ZkFold/Base/Protocol/Plonkup/PlonkConstraint.hs

```diff
@@ -40,48 +42,47 @@ instance (Arbitrary a, Finite a, ToConstant a Natural, KnownNat i) => Arbitrary
         qr <- arbitrary
         qo <- arbitrary
         qc <- arbitrary
-        x1 <- NewVar . toConstant @a <$> arbitrary
-        x2 <- NewVar . toConstant @a <$> arbitrary
-        x3 <- NewVar . toConstant @a <$> arbitrary
+        x1 <- Just . NewVar . toConstant @a <$> arbitrary
```

Comment:
It makes arbitrary less general than before. Can we include the possibility of constants as well?

Reply:
Are you talking about the case of `NewVar 0`? We can do it, but what should we set the probability of this to? Something like 1/20?

Reply:
Hmm, yes. I don't think it is possible to generate a constraint with less than 3 variables, like `x1 + x2 == 0`. I think equal probabilities for generating a constraint with `k` variables (k=1,2,3) would be great.